bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,253 @@
+import org.openhab.binding.pixometer.internal.serializer.CustomReadingInstanceDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link MeterHandler} is responsible for handling data and measurements of a meter thing
+ *
+ * @author Jerome Luckenbach - Initial contribution
+ */
+public class MeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeterHandler.class);
+
+    private static final String API_VERSION = "v1";
+    private static final String API_METER_ENDPOINT = "meters";
+    private static final String API_READINGS_ENDPOINT = "readings";
+
+    private final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeAdapter(ReadingInstance.class,
+            new CustomReadingInstanceDeserializer());
+    private final Gson gson = gsonBuilder.create();
+
+    private String resourceID;
+    private String meterID;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public MeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No handling needed, since we have only values to read.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Pixometer handler '{}'", getThing().getUID());
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            PixometerMeterConfiguration config = getConfigAs(PixometerMeterConfiguration.class);
+            setRessourceID(config.resourceId);
+
+            Bridge b = this.getBridge();
+            if (b == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "Could not find bridge (pixometer config). Did you choose one?");
+                return;
+            }
+
+            String token = new StringBuilder("Bearer ").append(((ApiserviceHandler) b.getHandler()).getAuthToken())
+                    .toString();
+
+            obtainMeterId(token);
+
+            // Start polling job with the interval, that has been set up in the bridge
+            int pollingPeriod = Integer.parseInt(b.getConfiguration().get(CONFIG_BRIDGE_REFRESH).toString());
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                logger.debug("Try to refresh meter data");
+                try {
+                    updateMeter(token);
+                } catch (RuntimeException r) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
+                }
+            }, 2, pollingPeriod, TimeUnit.MINUTES);
+            logger.debug("Refresh job scheduled to run every {} hours. for '{}'", pollingPeriod, getThing().getUID());
+        } catch (RuntimeException r) {
```

Comment:
can you be more specific?

Reply:
According to https://github.com/openhab/openhab2-addons/pull/4162#discussion_r347612348 i would remove it completely or just leave the try block with an empty finally statement.

Reply:
ok